### PR TITLE
Validate body parameters before making HTTP requests

### PR DESCRIPTION
**Pull request by pcothenet**
### What

- Validates body parameters before making the HTTP requests

### Why

- See [this PR](https://github.com/patch-technology/client-code-generation/pull/39) for details

### Notes

- #41 should be reviewed and merged first (this is built on top)

### SDK Release Checklist

- [x] Have you added an integration test for the changes?
- [x] Have you built the gem locally and made queries against it successfully?
- [x] Did you update the changelog?
- [x] Did you bump the package version?
- [ ] For breaking changes, did you plan for the release of the new SDK versions and deploy the API to production?

### CODE REVIEW

File: spec/integration/estimates_spec.rb [hunk 1]
```diff
@@ -30,6 +30,21 @@
...
+      destination_airport: "LAX"
+    )
+
+    flight_estimate_longer = Patch::Estimate.create_flight_estimate(
+      origin_airport: "SFO",
+      destination_airport: "JFK"
+    )
+
+    expect(flight_estimate.data.type).to eq 'flight'
+    expect(flight_estimate_longer.data.mass_g).to be > 2 * flight_estimate.data.mass_g
```
Comment (kleinjm) [inline spec/integration/estimates_spec.rb:45]: cool spec. I like that it uses known major US airports

File: lib/patch_ruby/api/estimates_api.rb [hunk 2]
```diff
@@ -35,8 +35,9 @@ def initialize(api_client = ApiClient.default)
     # @param create_bitcoin_estimate_request [CreateBitcoinEstimateRequest] 
     # @param [Hash] opts the optional parameters
     # @return [EstimateResponse]
-    def create_bitcoin_estimate(create_bitcoin_estimate_request, opts = {})
-      data, _status_code, _headers = create_bitcoin_estimate_with_http_info(create_bitcoin_estimate_request, opts)
+    def create_bitcoin_estimate(create_bitcoin_estimate_request = {}, opts = {})
+      _create_bitcoin_estimate_request = Patch::CreateBitcoinEstimateRequest.new(create_bitcoin_estimate_request)
```
Comment (kleinjm) [inline lib/patch_ruby/api/estimates_api.rb:39]: I'm sure you tested this locally but would it make sense to add a unit test around this as well to see some of these validations you're describing?

**Review by kleinjm — APPROVED** [review]
LGTM, just a suggestion on adding unit tests

File: lib/patch_ruby/api/estimates_api.rb (hunk 2 above)
Comment (pcothenet) [reply lib/patch_ruby/api/estimates_api.rb:39]: Good suggestion! Let me see how we could add that in!

**Pull request merged**